lib/mongoid/attributes/processing.rb

```diff
@@ -44,22 +44,46 @@ def process_attributes(attrs = nil)
       # @return [ true | false ] True if pending, false if not.
       def pending_attribute?(key, value)
         name = key.to_s
-
         aliased = if aliased_associations.key?(name)
           aliased_associations[name]
         else
           name
         end
-
         if relations.has_key?(aliased)
-          pending_relations[name] = value
+          set_pending_relation(name, aliased, value)
           return true
         end
         if nested_attributes.has_key?(aliased)
-          pending_nested[name] = value
+          set_pending_nested(name, aliased, value)
           return true
         end
-        return false
+        false
+      end
+
+      # Set value of the pending relation.
```

Comment:
Please improve this comment to explain what the method does and why, e.g. "checks if the name is a stored_as..."